pkg/kubelet/status/status_manager.go

```diff
@@ -296,13 +296,15 @@ func (m *manager) GetPodResizeStatus(podUID types.UID) v1.PodResizeStatus {
 func (m *manager) SetPodAllocation(pod *v1.Pod) error {
 	m.podStatusesLock.RLock()
 	defer m.podStatusesLock.RUnlock()
+
+	podAlloc := make(map[string]v1.ResourceRequirements)
+
 	for _, container := range pod.Spec.Containers {
 		alloc := *container.Resources.DeepCopy()
-		if err := m.state.SetContainerResourceAllocation(string(pod.UID), container.Name, alloc); err != nil {
```

Comment:
This change should stay in https://github.com/kubernetes/kubernetes/pull/129477#discussion_r1904749494. I don't think you need to touch `status_manager.go` in this PR